pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -58,7 +59,11 @@ func (a *Actuator) Exists(ctx context.Context, cluster *clusterv1.Cluster, machi
 	if err != nil {
 		return false, fmt.Errorf("failed to create scope for machine %q: %v", machine.Name, err)
 	}
-	defer scope.Close()
+	// The core machine controller calls exists() + create()/update() in the same reconciling operation.
+	// If exists() would store machineSpec/status object then create()/update() would still receive the local version.
+	// When create()/update() try to store machineSpec/status this might result in
+	// "Operation cannot be fulfilled; the object has been modified; please apply your changes to the latest version and try again."
+	// Therefore we don't close the scope here.
```

Comment:
The question then becomes: when is the scope closed? Can we document where/when?